Userland/Libraries/LibWeb/Painting/GradientPainting.cpp

```diff
@@ -272,11 +284,34 @@ void paint_conic_gradient(PaintContext& context, Gfx::IntRect const& gradient_re
     float start_angle = (360.0f - data.start_angle) + 90.0f;
     // Translate position/center to the center of the pixel (avoids some funky painting)
     auto center_point = Gfx::FloatPoint { position }.translated(0.5, 0.5);
+    // The flooring can make gradients that want soft edges look worse, so only floor if we have hard edges.
+    // Which makes sure the hard edge stay hard edges :^)
+    bool should_floor_angles = false;
+    auto& color_stops = data.color_stops.list;
+    for (size_t i = 0; i < color_stops.size() - 1; i++) {
+        if (color_stops[i + 1].position - color_stops[i].position <= 0.01f) {
+            should_floor_angles = true;
+            break;
+        }
+    }
     gradient_line.paint_into_rect(context.painter(), gradient_rect, [&](int x, int y) {
```

Comment:
We should probably capture by value here

Reply:
Why? The `paint_into_rect` (which is pretty much just loop) function is pretty much guaranteed to be inlined (since it's specialised with a unique lambda each time). So the actual closure should be optimised out I expect.

Reply:
I didn't look too far tbh.
The two reasons that cross my mind was:
- It's generally not good to take small types by reference (especially not a `const` one)
- If `paint_into_rect` come to save the lambda, the reference will outlive the value.

Definitely not a big issue though, feel free to ignore it.

Reply:
Yeah, `paint_into_rect` is just an iteration helper, it's an implementation detail in the same file. It's never going to save the lambda. I've now just marked it as `ALWAYS_INLINE` to make that clearer (the references here won't exist after it's inlined anyway).

Reply:
`paint_into_rect` being inlined does not ensure that the lambda will also be. It will probably, though.
And to be picky, there is no way to ensure that `paint_into_rect` will be inlined.